EIPS/eip-7933.md

```diff
@@ -0,0 +1,152 @@
+
+A more versatile way to encode intent (e.g., "deposit", "swap", "vote", etc.)
+
+Seamless QR-based user interactions without requiring wallet connection or dapp session
+
+By standardizing this flexible URI format, wallets can parse and act on requests directly from a link or QR code, enabling smoother real-world interactions and improving onboarding, security, and UX across Ethereum applications.
+
+## Specification
+
+The URI format extends the current `ethereum:` scheme:
+
+```
+request                 = schema_prefix [ rpc_method ] [ "-" target_address ] [ "@" chain_id ] [ "?" parameters ]
+schema_prefix           = "ethereum" ":" [ "pay-" / rpc_method "-" ]
+rpc_method             = STRING
+target_address         = ethereum_address / "CURRENT_ACCOUNT"
+chain_id               = 1*DIGIT
+ethereum_address       = ( "0x" 40*HEXDIG ) / ENS_NAME
+parameters             = parameter *( "&" parameter )
+parameter              = key "=" value
+key                    = STRING
+value                  = number / ethereum_address / STRING / base64_string / hex_value / "CURRENT_ACCOUNT"
+number                 = [ "-" / "+" ] *DIGIT [ "." 1*DIGIT ] [ ( "e" / "E" ) [ 1*DIGIT ] ]
+base64_string          = *( ALPHA / DIGIT / "+" / "/" ) [ "=" [ "=" ] ]
+hex_value              = "0x" 1*HEXDIG
+```
+
+`ethereum:<rpc_method>-0x0000000000000000000000000000000000000000@<chain_id>[\?<key>=<value>&...]`
+
+The special value `CURRENT_ACCOUNT` is a placeholder that will be replaced by the wallet implementation with the user's currently selected Ethereum address. When a wallet encounters this placeholder in the URI, it should substitute the user's current active address before processing the request. This allows URIs to be created that dynamically target the user's active wallet address without needing to know it in advance. The uppercase format follows common convention for symbolic constants.
+
+The zero address (`0x0000000000000000000000000000000000000000`) is used as a placeholder in URIs when calling RPC methods that don't require a target address. This maintains consistency with the URI format while allowing for RPC calls like `wallet_addEthereumChain` or `eth_chainId` that operate at the wallet/node level rather than targeting a specific contract. Wallets should ignore this zero address for methods where it's not relevant to the operation.
+
+### JSON-like Bracket Notation
+
+For parameters that represent nested objects or arrays, the URI format supports JSON-like bracket notation:
+
+- Objects: Use `[key]` to denote nested object properties
+- Arrays: Use `[index]` to denote array elements
+
+For example, a nested object parameter like:
+
+```json
+{
+  "chainId": "0x1",
+  "chainName": "Ethereum Mainnet",
+  "rpcUrls": ["https://mainnet.infura.io/v3/..."],
+  "iconUrls": ["https://example.com/icon.png"],
+  "nativeCurrency": {
+    "name": "Ether",
+    "symbol": "ETH",
+    "decimals": 18
+  },
+  "blockExplorerUrls": ["https://etherscan.io"]
```

Comment:
Avoid using real services, this URL can be example.com.

Reply:
ty